### ISSUE TITLE
Make common Python 2 and Python 3 compatible

### ISSUE THREAD

**Issue by jayich** [CampbellGroup/common#202]

**Comment by jayich** [comment]
treedict is not Python 3 compatible.

**Comment by fanmingyu212** [comment]
@aransfor I started a branch https://github.com/CampbellGroup/common/tree/mf/python_3_upgrading for python 2/3 compatibility. If using python 3, it will use pyqt5 and pyqt4 for python 2. I will first start upgrading the pulser and normalpmtflow.

**Comment by aransfor** [comment]
@fanmingyu212 , sounds good! How is it going to deal with pyqt4/pyqt5? Also do you know the status of pylabrad compatibility with Python3 or if anyone has fully switched?

**Comment by fanmingyu212** [comment]
@aransfor I planned to have a separate folder "client_py3" for python 3 clients. The servers would be compatible with python 2 and 3 simultaneously though. I just learned that the Martinis Group has switched to python 3. I think pylabrad is now compatible with python 3.6 (See https://github.com/labrad/pylabrad/blob/master/setup.py#L18).

**Comment by aransfor** [comment]
@fanmingyu212 are you guys up and running on Python3?

**Comment by fanmingyu212** [comment]
@aransfor Yes, we are. Most of the servers and clients we are using in common (pulser, script scanner, ixonServer) are compatible with python 3. I am working on upgrading all servers and clients that we use and will finish soon.

I am not able to upgrade all servers and clients in common to python 3 as we are not using some of them so we cannot test, but they should be similar to the other upgraded servers and clients.

**Comment by fanmingyu212** [comment]
General methods I used to port servers to python 3:
- run 2to3 tool to fix any syntax issues
- check str / bytes conversions (different for python 2 and 3 and 2to3 cannot find all of problems)
- labrad `settings` `"s"` parameter is string in both python 2 and 3, but it is byte string in python 2 and unicode string in python 3.

General methods for porting clients to python 3:
- Use PyQt5 instead of PyQt4. In PyQt5 many classes / modules are reordered, e.g. many `QtGui` submodules are now in `QtWidgets`.
- Similar with above about str / bytes.
